src/librustc_resolve/diagnostics.rs

```diff
@@ -444,6 +444,52 @@ impl SomeTrait for Foo { // ok!
 ```
 "##,
 
+E0406: r##"
+A function is referring to an associated type which hasn't been declared in
+the trait. Example of erroneous code:
+
+```compile_fail
+trait Foo {
+    type Bar;
+
+    // error: function signature contains a reference to `Self::Baz`, but
+    //        `Baz` hasn't been declared as an associated type of the trait
+    fn return_bool(&self, &Self::Bar, &Self::Baz) -> bool;
+}
+```
+
+The use of 'associated types' allows us declare variables which might be used
```

Comment:
No, that's not what I meant. Don't explain what the user tried to do but in which he/she did wrong.